go.mod

```diff
@@ -5,7 +5,7 @@ go 1.22.0
 require (
 	github.com/golangci/golangci-lint v1.50.1
 	github.com/google/go-cmp v0.6.0
-	github.com/hashicorp/terraform-plugin-framework v1.8.0
+	github.com/hashicorp/terraform-plugin-framework v1.11.0
```

Comment:
Embedded struct in a schema is supported from v1.11.0 (FYI: https://github.com/hashicorp/terraform-plugin-framework/releases/tag/v1.11.0).